### PR TITLE
add validation to prevent redirect loops and hops

### PR DESCRIPTION
**Pull request by asmega**

### CODE REVIEW

**Comment by tomas-stefano** [comment]
:+1:

File: app/models/redirect.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ class Redirect < ActiveRecord::Base
   validates :destination, presence: true, format: { with: /\A\/en|\/cy/ }
   validates :redirect_type, presence: true, inclusion: { in: REDIRECT_TYPES }
   validate  :validate_different_source_and_destination
+  validate  :destination_matches_existing_source
```
Comment (paulanthonywilson) [inline app/models/redirect.rb]: Am I reading the logic right? Shouldn't these be

```
validate :destination_does_not_match_existing_source
validate :source_does_not_match_existing_destination
```

Comment (asmega) [inline app/models/redirect.rb]: yep, i'll tweak that now

Comment (paulanthonywilson) [inline app/models/redirect.rb]: Cool, sorry for spotting it late.

**Pull request merged**